std/experimental/color/package.d

```diff
@@ -0,0 +1,305 @@
+// Written in the D programming language.
+
+/**
+    This package contains implementations of various common _color types.
+    Types are supplied for various _color spaces, along with comprehensive
+    _color space conversion functionality.
+
+    Authors:    Manu Evans
+    Copyright:  Copyright (c) 2015, Manu Evans.
+    License:    $(WEB boost.org/LICENSE_1_0.txt, Boost License 1.0)
+    Source:     $(PHOBOSSRC std/experimental/color/package.d)
+*/
+module std.experimental.color;
+
+public import std.experimental.color.rgb;
+import std.experimental.color.xyz : isXYZ, isxyY;
+
+@safe pure nothrow @nogc:
+
+
+/**
+Detect whether $(D T) is a color type defined under std.color.
+*/
+enum isColor(T) = isRGB!T || isXYZ!T || isxyY!T;
+
+///
+unittest
+{
+    static assert(isColor!RGB8 == true);
+    static assert(isColor!(XYZ!float) == true);
+    static assert(isColor!float == false);
+}
+
+
+// declare some common color types
+/** 24 bit RGB color type with 8 bits per channel. */
+alias RGB8 =    RGB!("rgb", ubyte);
+/** 32 bit RGB color type with 8 bits per channel. */
+alias RGBX8 =   RGB!("rgbx", ubyte);
+/** 32 bit RGB + alpha color type with 8 bits per channel. */
+alias RGBA8 =   RGB!("rgba", ubyte);
+
+/** Floating point RGB color type. */
+alias RGBf32 =  RGB!("rgb", float);
+/** Floating point RGB + alpha color type. */
+alias RGBAf32 = RGB!("rgba", float);
+
+/** 24 bit BGR color type with 8 bits per channel. */
+alias BGR8 =    RGB!("bgr", ubyte);
+/** 32 bit BGR color type with 8 bits per channel. */
+alias BGRX8 =   RGB!("bgrx", ubyte);
+/** 32 bit BGR + alpha color type with 8 bits per channel. */
+alias BGRA8 =   RGB!("bgra", ubyte);
+
+/** 8 bit luminance-only color type. */
+alias L8 =      RGB!("l", ubyte);
+/** 8 bit alpha-only color type. */
+alias A8 =      RGB!("a", ubyte);
+/** 16 bit luminance + alpha color type with 8 bits per channel. */
+alias LA8 =     RGB!("la", ubyte);
+
+/** 16 bit signed UV color type with 8 bits per channel. */
+alias UV8 =     RGB!("rg", byte);
+/** Floating point UV color type. */
+alias UVf32 =   RGB!("rg", float);
+
```

Comment:
I'm not sure about there being too many. Maybe adding them at this stage is premature.
But I do not believe that people should have to alias for the common ones.

I can't really argue for 32/64bit versions only 8/16bit ones. Since e.g. PNG supports up to 16bit.